src/database-sqlite3.cpp

```diff
@@ -136,6 +137,23 @@ void Database_SQLite3::endSave() {
 	SQLRES(sqlite3_step(m_stmt_end), SQLITE_DONE,
 		"Failed to commit SQLite3 transaction");
 	sqlite3_reset(m_stmt_end);
+
+	int total_size;
+	int checkpointed;
+	SQLOK(sqlite3_wal_checkpoint_v2(m_database, NULL,
+			SQLITE_CHECKPOINT_PASSIVE, &total_size, &checkpointed),
+		"failed to checkpoint database");
+	if (total_size >= 0 && checkpointed >= 0) {
+		int new_wal_backlog = total_size - checkpointed;
+		// Normally, sqlite would checkpoint every 1000 pages.
+		if (m_wal_backlog / 10000 < new_wal_backlog / 10000)
```

Comment:
This seems like a rather confusing way to write what is mostly essentially equivalent to `(new_wal_backlog > m_wal_backlog + 10000)`.
Is there a reason to round to 10000 frames?